http/httpclient/client.go

```diff
@@ -127,6 +125,15 @@ func (jc *HttpClient) Send(method, url string, content []byte, followRedirect, c
 	return
 }
 
+func (jc *HttpClient) createReq(method, url string, content []byte) (req *http.Request, err error) {
+	log.Debug(fmt.Sprintf("Sending HTTP %s request to: %s", method, url))
+	if content != nil {
+		return jc.newRequest(method, url, bytes.NewBuffer(content))
+	}
+	req, err = jc.newRequest(method, url, nil)
+	return req, errorutils.CheckError(err)
```

Comment:
The CheckError should probably not be here, but moved into the ```jc.newRequest``` method.
Note that due to the fact that the error isn't checked inside ```jc.newRequest```, the error returned in line 131 is never checked.
After we make the above change, we should also look at all the usages of ```jc.newRequest``` and remove the CheckError's from them.